clusterDeployer.py

```diff
@@ -571,13 +571,15 @@ def wait_for_workers(self) -> None:
         lh = host.LocalHost()
         bf_workers = [x for x in self._cc.workers if x.kind == "bf"]
         connections: dict[str, host.Host] = {}
+        last_msg = None
         for try_count in itertools.count(0):
             workers = [w.name for w in self._cc.workers]
             n_not_ready_workers = sum(1 for w in workers if not self.client().is_ready(w))
             if n_not_ready_workers == 0:
                 break
 
-            logger.info(f"Not all workers ready (try #{try_count}). {n_not_ready_workers} are not ready yet.")
+            last_msg = logger.info_once(last_msg, f"Not all workers ready. {n_not_ready_workers} are not ready yet.")
```

Comment:
Can you make it so that info_once stores the last message?

Reply:
That adds mutable state to the logger. That would require making it thread-local, since the logging is used by multiple threads.

It also means, that you cannot interleave different `info_once()`, since there is only one state (which is either global/bad or thread-local/OK).

I don't think that's better, because it's less flexible while hiding how things work (for things that are easy to understand).

Anyway. Will change it to have a thread-local state.